app/controllers/hashtags_controller.rb

```diff
@@ -3,13 +3,16 @@
 class HashtagsController < ApplicationController
   requires_login
 
+  HASHTAGS_PER_MINUTE = 120
+
   def show
     raise Discourse::InvalidParameters.new(:slugs) if !params[:slugs].is_a?(Array)
 
     all_slugs = []
     tag_slugs = []
 
     params[:slugs].each do |slug|
+      RateLimiter.new(current_user, "hashtags-min", HASHTAGS_PER_MINUTE, 1.minute).performed!
```

Comment:
A rate limit is kinda like using a rocket launcher to kill a fly.

I was thinking of just testing the size of `params[:slug]` and limiting it to a "decent" value. (just because I don't trust the "user")